app.js

```diff
@@ -0,0 +1,217 @@
+      const color = event.target.dataset.color;
+      if (color) {
+        this.editNoteColor(color);
+      }
+    });
+
+    this.$form.addEventListener("submit", (event) => {
+      event.preventDefault();
+      const title = this.$noteTitle.value;
+      const text = this.$noteText.value;
+      const hasNote = title || text;
+      if (hasNote) {
+        this.addNote({ title, text });
+      }
+    });
+
+    this.$closeBtn.addEventListener("click", (event) => {
+      event.stopPropagation();
+      this.closeForm();
+    });
+
+    this.$modalCloseBtn.addEventListener("click", (event) => {
+      this.modalClose(event);
+    });
+  }
+
+  handleFormClick(event) {
+    const isFormClicked = this.$form.contains(event.target);
+
+    const title = this.$noteTitle.value;
+    const text = this.$noteText.value;
+    const hasNote = title || text;
+
+    if (isFormClicked) {
+      this.openForm();
+    } else if (hasNote) {
+      this.addNote({ title, text });
+    } else {
+      this.closeForm();
+    }
+  }
+
+  openForm() {
+    this.$form.classList.add("form-open");
+    this.$noteTitle.style.display = "block";
+    this.$formBtns.style.display = "block";
+  }
+
+  closeForm() {
+    this.$form.classList.remove("form-open");
+    this.$noteTitle.style.display = "none";
+    this.$formBtns.style.display = "none";
+    this.$noteTitle.value = "";
+    this.$noteText.value = "";
+  }
+
+  addNote({ title, text }) {
+    const newNote = {
+      title,
+      text,
+      color: "white",
+      id: this.notes.length > 0 ? this.notes[this.notes.length - 1].id + 1 : 1,
+    };
+    this.notes = [...this.notes, newNote];
+    this.saveNotes();
+    this.showNotes();
+    this.closeForm();
+  }
+
+  modalOpen(event) {
+    if (event.target.matches(".fa-trash")) return;
+
+    if (event.target.closest(".display-note")) {
+      this.$modalSection.classList.toggle("open-modal");
+      this.$modalTitle.value = this.title;
+      this.$modalText.value = this.text;
+    }
+  }
+
+  modalClose(event) {
+    this.editNote(event);
+    this.$modalSection.classList.toggle("open-modal");
+  }
+
+  editNote() {
+    const title = this.$modalTitle.value;
+    const text = this.$modalText.value;
+    this.notes = this.notes.map((note) =>
+      note.id === Number(this.id) ? { ...note, title, text } : note
+    );
+    this.saveNotes();
+    this.showNotes();
+  }
+
+  selectNote(event) {
+    const $selectedNote = event.target.closest(".display-note");
+    if (!$selectedNote) return;
+    const [$noteTitle, $noteText] = $selectedNote.children;
+    this.title = $noteTitle.innerText;
+    this.text = $noteText.innerText;
+    this.id = $selectedNote.dataset.id;
+  }
+
+  openTooltip(event) {
+    if (!event.target.matches(".fa-palette")) return;
+    this.id = event.target.nextElementSibling.dataset.id;
+    const noteCoords = event.target.getBoundingClientRect();
+    const horizontal = noteCoords.left
```

Comment:
this line defines the position of your tooltip you can try to add or reduce different pixel values here